### PR TITLE
Remove warning filters that are no longer needed for testing

### PR DESCRIPTION
**Pull request by rosteen**
Does what it says on the tin.

### CODE REVIEW

**Comment by rosteen** [comment]
Marking this trivial since the only requirement is that the tests pass.

Not sure how this affected coverage, I think Codecov is hallucinating.

File: pyproject.toml [hunk 1]
```diff
@@ -127,26 +127,12 @@ text_file_format = "rst"
 addopts = "--doctest-rst --import-mode=append"
 filterwarnings = [
     "error",
-    "ignore:numpy\\.ufunc size changed:RuntimeWarning",
```
Comment (pllim) [inline pyproject.toml:130]: Need to keep the ufunc one. It comes up once in a while depending on how CI installed stuff and in what order.

**Comment by pllim** [comment]
I think you removed too many...

```
E   DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated
and scheduled for removal in a future version.
Use timezone-aware objects to represent datetimes in UTC:
datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
```

**Comment by rosteen** [comment]
> I think you removed too many...

Blast. I'll come back to this tomorrow, I was testing locally but didn't have all the dev dependencies.

**Comment by pllim** [comment]
`datetime.datetime.utcfromtimestamp()` is actually from Python 3.12 but yeah. 😬 

Thanks for tackling this!

**Comment by rosteen** [comment]
The 3.12 dev test now hits a real error from `pandas` - my attempt at getting a local env set up with all the dev requirements was a mess, so I can't 100% confirm this is good to go. I'll see if https://github.com/pandas-dev/pandas/pull/55707 gets merged quickly to resolve the `pandas` failure and put this on the back-burner until next week.

**Comment by pllim** [comment]
Yeah probably save us time to just wait out the *panda*monium.

**Comment by mroeschke** [comment]
Just merged in https://github.com/pandas-dev/pandas/pull/55707 so hopefully this should show up in tomorrow's wheels

**Review by pllim — APPROVED** [review]
devdeps is green now. LGTM. Thanks for knocking out a bunch of these ignores!

**Pull request merged**